icicle/appUtils/msm/msm.cu

```diff
@@ -338,59 +340,63 @@ namespace msm {
     // this function computes msm using the bucket method
     template <typename S, typename P, typename A>
     cudaError_t bucket_method_msm(
-      int bitsize,
-      int c,
+      unsigned bitsize,
+      unsigned c,
       S* scalars,
       A* points,
-      int size,
+      unsigned batch_size,
```

Comment:
The definition of these 3 variables is unclear, please add a description. Also, it seems that only 2 out of 3 are needed no?

Reply:
I don't think 2 variables are enough:
- msm_size is the number of elements per msm sum
- batch_size is how many msms to compute in the batch
- nof_points can either be msm_size (if sharing points) or msm_size*batch_size (if not sharing points).

maybe I can replace nof_points by a boolean that tells if sharing points between batch elements or not. It would still be 3 variables and require modification of 'MSMConfig' which would break the API so I prefer avoiding it.

Reply:
I would leave three variables. By design it should be possible to have different points in the first and second MSMs and then re-use points from the first for the third and then points from the second for the fourth and so on...